src/relay/backend/interpreter.cc

```diff
@@ -39,9 +42,81 @@
 namespace tvm {
 namespace relay {
 
-using namespace runtime;
+using runtime::ADT;
+using runtime::ADTObj;
+using runtime::NDArray;
+using runtime::TVMArgsSetter;
+using runtime::operator<<;
+
+namespace {
+// TODO(mbs): Centralize.
+struct PairHash {
+  template <typename T1, typename T2>
+  std::size_t operator()(const std::pair<T1, T2>& k) const {
+    return std::hash<T1>()(k.first) ^ std::hash<T2>()(k.second);
+  }
+};
+
+// Analogue of FlattenTupleType for runtime ADT vs NDArray values.
+// TODO(mbs): Hoist somewhere sensible, maybe op/memory.h?
+void FlattenADTAux(const ObjectRef& object_ref, std::vector<NDArray>* out) {
```

Comment:
Yeah can we maybe just add an issue to unify these.

Reply:
done